src/app/Main/Body/QueryPage/Results/Content-Results.component.tsx

```diff
@@ -1,126 +1,125 @@
+
     const rowAttribute: FilterType = table.row;
     const colAttribute: FilterType = table.column;
-
-    const { t } = useTranslation(["QueryPage"]);
+    const isCol: boolean = colAttribute !== "";
+    const isRow: boolean = rowAttribute !== "";
+    const state = { isCol, isRow };
 
     const getTableContext = async (
-        dataToCountIsolates: DbCollection
+        isolateCountProp: IsolateCountedDTO
     ): Promise<void> => {
-        const rowValues = getValuesOfOneFilterAttribute(
-            rowAttribute,
-            data.uniqueValues,
-            filter.selectedFilter
-        );
-        const colValues = getValuesOfOneFilterAttribute(
-            colAttribute,
-            data.uniqueValues,
-            filter.selectedFilter
-        );
-        const nrOfSelectedIsolates = dataToCountIsolates.length;
-        if (rowAttribute.length !== 0 && colAttribute.length !== 0) {
-            const rowsWithIsolates = generateRowsWithIsolates(
-                dataToCountIsolates,
-                colAttribute,
-                rowAttribute,
-                rowValues,
-                colValues,
-                "both"
-            );
-            const rowsWithPercent = calculateRowsWithPercent(rowsWithIsolates, nrOfSelectedIsolates)
-            setState({
-                isCol: true,
-                isRow: true,
-            });
-            setColumnAttributes(colValues);
+        if (!isCol && !isRow) {
             setTable({
                 ...table,
-                statisticDataAbsolute: rowsWithIsolates,
-                statisticDataPercent: rowsWithPercent,
+                statisticDataAbsolute: [],
             });
-        } else if (rowAttribute.length === 0 && colAttribute.length !== 0) {
-            const rowsWithIsolates = generateRowsWithIsolates(
-                dataToCountIsolates,
-                colAttribute,
-                rowAttribute,
-                [t("Results.TableHead")],
-                colValues,
-                "col"
-            );
-            const rowsWithPercent = calculateRowsWithPercent(rowsWithIsolates, nrOfSelectedIsolates)
-            setState({
-                isCol: true,
-                isRow: false,
+        } else {
+            const rowValues: string[] = !isRow
+                ? [t("Results.TableHead")]
+                : (_.isEmpty(filter.selectedFilter[rowAttribute])
+                ? data.uniqueValues[rowAttribute]
+                : filter.selectedFilter[rowAttribute])
+
+            const colValues: string[] = !isCol
+                ? [t("Results.TableHead")]
+                : (_.isEmpty(filter.selectedFilter[colAttribute])
+                ? data.uniqueValues[colAttribute]
+                : filter.selectedFilter[colAttribute])
+
+            const isolateColValues: string[] = [t("Results.TableHead")];
+            const isolateRowValues: string[] = [t("Results.TableHead")];
+            isolateCountProp.groups.forEach((isolateCount) => {

```

Comment:
i think this whole algorithm can be simplified. E.g. you iterate over isolateCountGroups here and again for each rowValue and colValue pair below. I don't exactly what the algorithm does (and have no time to get into it), so i can give no tip how to make it better. But if you have the time you can refactor it. Or we can discuss this over jitsi, so you can explain what the algo should do.

Reply:
I think it makes no sense to start the iteration over the isolateCountGroups, because in each iteration a row is created for the table. So I need one iteration for each rowValue - that's why I start with that.

Reply:
Discussed in meeting